tests/unittests/CoreText/CTLineTests.mm

```diff
@@ -257,4 +257,31 @@
     line = CTLineCreateWithAttributedString(string);
     EXPECT_EQ(9, CTLineGetGlyphCount(line));
     CFRelease(line);
+}
+
+TEST(CTLine, CTLineGetTypographicBounds) {
+    UIFontDescriptor* fontDescriptor = [UIFontDescriptor fontDescriptorWithName:@"Segoe UI" size:40];
+    CTFontRef font = (__bridge CTFontRef)[UIFont fontWithDescriptor:fontDescriptor size:40];
+    CFAttributedStringRef baseString = (__bridge CFAttributedStringRef)getAttributedString(@"x");
+    CFAttributedStringRef variableString = (__bridge CFAttributedStringRef)getAttributedString(@"gH");
+    CTLineRef baseLine = CTLineCreateWithAttributedString(baseString);
+    CFAutorelease(baseLine);
+    CTLineRef variableLine = CTLineCreateWithAttributedString(variableString);
+    CFAutorelease(variableLine);
+
+    CGFloat baseAscent, variableAscent, baseDescent, variableDescent, baseLeading, variableLeading;
+    CGFloat baseWidth = CTLineGetTypographicBounds(baseLine, &baseAscent, &baseDescent, &baseLeading);
+    EXPECT_LT(0, baseWidth);
+    EXPECT_EQ_MSG(CTFontGetLeading(font), baseLeading, "Leading should always be the same as the font");
```

Comment:
According to their documentation on their typography stuff it should, and it has from what I've seen testing.